src/adservice/src/main/java/oteldemo/AdService.java

```diff
@@ -201,7 +201,8 @@ public void getAds(AdRequest req, StreamObserver<AdResponse> responseObserver) {
             Attributes.of(
                 adRequestTypeKey, adRequestType.name(), adResponseTypeKey, adResponseType.name()));
 
-        if (ffClient.getBooleanValue(ADSERVICE_FAILURE, false, evaluationContext)) {
+        // Throw 1/10 of the time to simulate a failure when the feature flag is enabled
+        if (ffClient.getBooleanValue(ADSERVICE_FAILURE, false, evaluationContext) && random.nextInt(10) == 0) {
```

Comment:
Is it a good practice to have the flag percentage in the service code?

I know we had that before, but I believe that with flagd we can now have a better approach.

Reply:
It depends on whether you want to be able to change the failure rate at an arbitrary value at runtime. The docs mention that's a static 1/10 failure rate and that's easier to enforce in code. As weird as it sounds, the failure condition is basically business logic in this app.